diskfs.go

```diff
@@ -213,17 +212,10 @@ func initDisk(f *os.File, openMode OpenModeOption, sectorSize SectorSize) (*disk
 	case mode&os.ModeDevice != 0:
 		log.Debug("initDisk(): block device")
 		diskType = disk.Device
-		file, err := os.Open(f.Name())
+		size, err = getBlockDeviceSize(f)
 		if err != nil {
-			return nil, fmt.Errorf("error opening block device %s: %s", f.Name(), err)
+			return nil, fmt.Errorf("error getting block device %s size: %s", f.Name(), err)
 		}
-		defer file.Close()
-		size, err = file.Seek(0, io.SeekEnd)
-		if err != nil {
-			return nil, fmt.Errorf("error seeking to end of block device %s: %s", f.Name(), err)
-		}
-		lblksize, pblksize, err = getSectorSizes(f)
```

Comment:
So how are logical block size and physical block size (`lblksize` and `pblksize`, respectively) supposed to be set?

Reply:
It looks very much like a big mistake 🤔. It also means that my manual tests were not relevant…

Reply:
Fixed and re-tested on Mac VM & Linux loopback